modules/justfiles/justfiles.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+get_yaml_array CONFIG_SELECTION '.include[]' "$1"
+VALIDATE="$(echo "$1" | yq -I=0 ".validate")"
+
+IMPORT_FILE="/usr/share/ublue-os/just/60-custom.just"
+CONFIG_FOLDER="${CONFIG_DIRECTORY}/justfiles"
+DEST_FOLDER="/usr/share/bluebuild/justfiles"
+
+# Abort if justfiles folder is not present
+if [ ! -d "${CONFIG_FOLDER}" ]; then
+    echo "Error: The config folder '${CONFIG_FOLDER}' was not found."
+    exit 1
+fi
+
+# Include all files in the folder if none specified
+if [[ ${#CONFIG_SELECTION[@]} == 0 ]]; then
+    CONFIG_SELECTION=($(ls "${CONFIG_FOLDER}"))
```

Comment:
This should account for cases when there are folders present with justfiles included, but not specified in recipe.

EDIT: Edited the suggestion to mimic the `ls` behavior without its drawbacks

```suggestion
    CONFIG_SELECTION=($(find "${CONFIG_FOLDER}" -mindepth 1 -maxdepth 1 -exec basename {} \;))
```

Reply:
I wouldn't do this. It will work but treat every file in the same folder separately. I'm bad at explaining myself, so I'll just show you.

With the original code:

![Snímek obrazovky z 2024-06-26 10-51-28](https://github.com/blue-build/modules/assets/136916485/bfca6baa-3d0d-4e8d-af50-f157d939875f)

With your proposition:

![Snímek obrazovky z 2024-06-26 10-50-26](https://github.com/blue-build/modules/assets/136916485/c7c63f9b-23f4-4ff3-8a3f-3c5c2e8f9969)

Reply:
Regardless, the original code goes against the golden rule; *Never parse the output of `ls`.*
Globs exist, if possible, use them.

Reply:
I'm not sure what to do here, that part is only supposed to fill the variable with everything that is inside the CONFIG_FOLDER and then for every file/folder, all justfiles are found, copied, imported. I don't like what's happening in the second screenshot.

If I used something like `CONFIG_SELECTION=($(echo "${CONFIG_FOLDER}"/*))`, is that equally as bad ?

Reply:
My proposal is indeed not ideal in this case. I will think about the another one.

Reply:
I added the suggestion which just mimics the `ls` behavior without its drawbacks

Reply:
> I edited the suggestion which just mimics the `ls` behavior without its drawbacks

That **works**, but why did you put the `))` on the next line ?

Also, would my previous suggestion: 
```
CONFIG_SELECTION=($(echo "${CONFIG_FOLDER}"/* | sed "s|${CONFIG_FOLDER}/||g"))`
```
be bad in any way ? I'm just curious. Throw some kind of documentation at me if possible please.

Reply:
> That **works**, but why did you put the `))` on the next line ?

Mistake in editing, fixed.

> Also, would my previous suggestion:
> 
> ```
> CONFIG_SELECTION=($(echo "${CONFIG_FOLDER}"/* | sed "s|${CONFIG_FOLDER}/||g"))`
> ```
> 
> be bad in any way ? I'm just curious. Throw some kind of documentation at me if possible please.

echo treates `*` as a string, not as a glob, so it wouldn't work.